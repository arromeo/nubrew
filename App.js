import React from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { AppLoading, Asset, Font, Icon } from 'expo';
import AppNavigator from './navigation/AppNavigator';
const port = require('./dev_port.json');

export default class App extends React.Component {
  state = {
    isLoadingComplete: false,
    favorites: null,
    user_id: 1
  };

  componentDidMount() {
    let url = `${port.DEV_PORT}/api/user/${this.state.user_id}/favorites`;
    fetch(url)
      .then(res => res.json())
      .then(data => {
        this.setState ({ favorites: data.result, })
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <View style={styles.container}>
          {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
          <AppNavigator screenProps={{user_id: this.state.user_id, favorites: this.state.favorites }}/>
        </View>
      );
    }
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/robot-dev.png'),
        require('./assets/images/robot-prod.png'),
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
      }),
    ]);
  };

  _handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});
