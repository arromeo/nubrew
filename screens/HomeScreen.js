// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import {
  Image,
  ScrollView,
  StyleSheet,
  Text,
  View,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { WebBrowser } from 'expo';
import { MonoText } from '../components/StyledText';

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      event: null,
      recommendedBeer: null,
      loading: true,
    }
  }
  static navigationOptions = {
    title: 'NuBrew',
  };

  randomGen = (data) => {
    return Math.floor(Math.random() * data.length);
  }

  componentDidMount() {
    fetch(`${port.DEV_PORT}/api/index`)
      .then(res => res.json())
      .then(data => this.setState ({
        event: data.result.events,
        recommendedBeer: data.result.featured_beer[this.randomGen(data.result.featured_beer)],
        loading: false
      }))
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    const event = this.state.event;
    const recommendedBeer = this.state.recommendedBeer;
    const { navigate } = this.props.navigation;

    return (
      <View style={styles.container}>
        {this.state.loading && 
          <View><Text>LoadingScreen goes here</Text></View>
        }
        {!this.state.loading && 
          <ScrollView style={styles.container}>
            <View style={styles.contentContainer}>

              <FlatList
                data={event}
                keyExtractor={item => item.id.toString()}
                renderItem={({item}) => 
                <TouchableOpacity
                  style={[styles.eventContainer, styles.homeScreenFilename]}
                  onPress={() => {
                    navigate({
                      routeName: 'Find',
                      params: {
                        type: 'Event',
                        item
                      }
                    });}}
                  >
                  <View style={[styles.eventDetailsContainer, styles.homeScreenFilename]}>
                    <Text>{item.event_name} at {item.store_name}</Text>
                    <MonoText style={styles.codeHighlightText}>{item.details.split('').slice(0, 60).join("")}...</MonoText>
                  </View>
                </TouchableOpacity>
              }/>
            </View>

            <View style={styles.contentContainer}>
              <TouchableOpacity 
                style={[styles.recommendationContainer, styles.homeScreenFilename]}
                onPress={() => {
                  navigate({
                    routeName: 'Find',
                    params: {
                      type: 'Beer',
                      recommendedBeer
                    }
                  });}}
                >
                <Image
                  source={
                    __DEV__
                      ? require('../assets/images/robot-dev.png')
                      : require('../assets/images/robot-prod.png')
                  }
                  size={100}
                />
                <Text>{recommendedBeer.brewery_name}'s</Text>
                <Text>{recommendedBeer.beer_name}</Text>
                <MonoText style={styles.codeHighlightText}>IBU: {recommendedBeer.ibu}</MonoText>
                <MonoText style={styles.codeHighlightText}>ABV: {recommendedBeer.abv}</MonoText>
              </TouchableOpacity>
      
              <TouchableOpacity 
                style={[styles.recommendationContainer, styles.homeScreenFilename]}
                onPress={() => {
                  navigate({
                    routeName: 'Find',
                    params: {
                      type: 'CrowdRecommendations',
                    }
                  });}}
              >
                <Ionicons name="md-medal" size={100} color="black"/>
                <Text>Crowd</Text>
                <Text>Recommendations</Text>
              </TouchableOpacity>
            </View>

          </ScrollView>
        }

      </View>
    );
  }

  _handleLearnMorePress = () => {
    WebBrowser.openBrowserAsync('https://docs.expo.io/versions/latest/guides/development-mode');
  };

  _handleHelpPress = () => {
    WebBrowser.openBrowserAsync(
      'https://docs.expo.io/versions/latest/guides/up-and-running.html#can-t-see-your-changes'
    );
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  contentContainer: {
    paddingTop: 15,
    paddingBottom: 15,
    borderWidth: 0.5,
    borderColor: "black",
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "row",
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
  eventContainer: {
    flex: 1,
    flexDirection: "row",
  },
  eventDetailsContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
  },
  recommendationContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    width: '45%',
    margin: 10,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  bottomOptionsContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    width: '28.33%',
    margin: 10,
  },
  welcomeImage: {
    width: 50,
    height: 40,
    resizeMode: 'contain',
    marginTop: 3,
    marginLeft: -10,
  },
  eventImage: {
    width: 100,
    height: 80,
    resizeMode: 'contain',
    margin: 10,
    width: '20%',
  },
  homeScreenFilename: {
    marginVertical: 7,
  },
  optionScreenFilename: {
    marginVertical: 7,
  },
  getStartedText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    lineHeight: 24,
    textAlign: 'center',
  },
});
