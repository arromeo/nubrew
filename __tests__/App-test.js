import 'react-native';
import React from 'react';
import App from '../App';
import HomeScreen from '../screens/HomeScreen';
import renderer from 'react-test-renderer';

describe('App snapshot', () => {

  it('renders the loading screen', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the root without loading screen', async () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the homescreen', async () => {
    const tree = renderer.create(<HomeScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
