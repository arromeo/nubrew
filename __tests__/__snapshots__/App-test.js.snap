// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "paddingTop": 64,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            collapsable={false}
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#61170E",
                  "elevation": 4,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 0.5,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 0.5,
                  "zIndex": 1,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffeb3b",
                      "bottom": 0,
                      "height": 2,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "overflow": "scroll",
                  }
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "orange",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            onLayout={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "backgroundColor": "orange",
                "borderBottomColor": "#A7A7AA",
                "borderBottomWidth": 0.5,
                "height": 64,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "flexDirection": "row",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 28,
                          },
                          Object {
                            "color": "#61170E",
                            "marginLeft": 10,
                          },
                          Object {
                            "fontFamily": "ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï‡™
                    </Text>
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "marginLeft": 10,
                        }
                      }
                    >
                      Home
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
